vnext/PropertySheets/ManagedCodeGen/Microsoft.ReactNative.Managed.CodeGen.targets

```diff
@@ -0,0 +1,107 @@
+    </MSBuild>
+
+    <MSBuild Projects="$(_ReactNativeCodeGenProjectPath)" Targets="GetPublishedToolPath" BuildInParallel="$(BuildInParallel)" Properties="$(_ReactNativeCodeGenProjectProperties)" ContinueOnError="!$(BuildingProject)" RemoveProperties="%(_MSBuildProjectReferenceExistent.GlobalPropertiesToRemove)">
+
+      <!-- This target will return the tools executable. We unfortunately have ot use a separate target invocation
+         - because msbuid does not allow returning multiple pieces of data from a single invocation nor is it easy
+         - to setup a convention where the first element would be the executable
+         -->
+      <Output TaskParameter="TargetOutputs" ItemName="_ReactNativeCodeGenToolExecutableTemp" />
+
+    </MSBuild>
+
+    <ItemGroup>
+      <!-- MsBuild does not allow targets to return properties, so map it to a property from an itemgroup -->
+      <_ReactNativeCodeGenToolFiles Include="@(_ReactNativeCodeGenToolFilesTemp)" />
+    </ItemGroup>
+
+    <PropertyGroup>
+      <!-- MsBuild does not allow targets to return properties, so map it to a property from an itemgroup -->
+      <_ReactNativeCodeGenToolExecutable>@(_ReactNativeCodeGenToolExecutableTemp)</_ReactNativeCodeGenToolExecutable>
+    </PropertyGroup>
+
+  </Target>
+
+  <Target 
+    Name="ReactNativeManagedCodeGen" 
+    DependsOnTargets="_ReactNativeManagedCodeGenPublishTool" 
+    Inputs="$(MSBuildProjectFullPath);@(Compile);@(_ReactNativeCodeGenToolFiles)" 
+    Outputs="$(ReactNativeCodeGenFile);$(_ReactNativeCodeGenResponseFile)"
+    Condition="'$(DesignTimeBuild)' != 'true'">
+
+    <ItemGroup>
+      <_ReactNativeCodeGenDefineConstatnsAsItem Include="$(DefineConstants)" />
+    </ItemGroup>
+
+    <ItemGroup>
+      <_ReactNativeCodeGenResponseFileLines Include="/ns:$(RootNamespace)" />
+      <_ReactNativeCodeGenResponseFileLines Include=" " />
+      <_ReactNativeCodeGenResponseFileLines Include="/o:$(ReactNativeCodeGenFile)" />
+      <_ReactNativeCodeGenResponseFileLines Include=" " />
+      <_ReactNativeCodeGenResponseFileLines Include="# Defines" />
+      <_ReactNativeCodeGenResponseFileLines Include="@(_ReactNativeCodeGenDefineConstatnsAsItem->'/d:%(Identity)')" />
+      <_ReactNativeCodeGenResponseFileLines Include=" " />
+      <_ReactNativeCodeGenResponseFileLines Include="# Source Files" />
+      <_ReactNativeCodeGenResponseFileLines Include="@(Compile->'/s:%(fullpath)')" />
+      <_ReactNativeCodeGenResponseFileLines Include=" " />
+      <_ReactNativeCodeGenResponseFileLines Include="# References" />
+      <_ReactNativeCodeGenResponseFileLines Include="@(ReferencePath->'/r:%(fullpath)')" />
+    </ItemGroup>
+
+    <MakeDir Directories="$(XamlGeneratedOutputPath)intermediatexaml\" />
+
+    <!-- Write the response file -->
+    <WriteLinesToFile File="$(_ReactNativeCodeGenResponseFile)" Lines="@(_ReactNativeCodeGenResponseFileLines)" Overwrite="true" Encoding="Unicode"/>
```

Comment:
Overwrite only if different? to avoid rebuilds

Reply:
good point, was left over from debugging stuff..